src/Workspaces/Core/Portable/Workspace/Solution/SourceGeneratedDocumentIdentity.cs

```diff
@@ -41,31 +42,30 @@ public static SourceGeneratedDocumentIdentity Generate(ProjectId projectId, stri
         // ensure we don't have collisions.
         var generatorIdentity = SourceGeneratorIdentity.Create(generator, analyzerReference);
 
-        // Combine the strings together; we'll use Encoding.Unicode since that'll match the underlying format; this can be made much
-        // faster once we're on .NET Core since we could directly treat the strings as ReadOnlySpan<char>.
-        var projectIdBytes = projectId.Id.ToByteArray();
-
         // The assembly path should exist in any normal scenario; the hashing of the name only would apply if the user loaded a
         // dynamic assembly they produced at runtime and passed us that via a custom AnalyzerReference.
         var assemblyNameToHash = generatorIdentity.AssemblyPath ?? generatorIdentity.AssemblyName;
 
-        using var _ = ArrayBuilder<byte>.GetInstance(capacity: (assemblyNameToHash.Length + 1 + generatorIdentity.TypeName.Length + 1 + hintName.Length) * 2 + projectIdBytes.Length, out var hashInput);
-        hashInput.AddRange(projectIdBytes);
-
-        // Add a null to separate the generator name and hint name; since this is effectively a joining of UTF-16 bytes
-        // we'll use a UTF-16 null just to make sure there's absolutely no risk of collision.
-        hashInput.AddRange(Encoding.Unicode.GetBytes(assemblyNameToHash));
-        hashInput.AddRange(0, 0);
-        hashInput.AddRange(Encoding.Unicode.GetBytes(generatorIdentity.TypeName));
-        hashInput.AddRange(0, 0);
-        hashInput.AddRange(Encoding.Unicode.GetBytes(hintName));
-
-        // The particular choice of crypto algorithm here is arbitrary and can be always changed as necessary. The only requirement
-        // is it must be collision resistant, and provide enough bits to fill a GUID.
-        using var crytpoAlgorithm = System.Security.Cryptography.SHA256.Create();
-        var hash = crytpoAlgorithm.ComputeHash(hashInput.ToArray());
-        Array.Resize(ref hash, 16);
-        var guid = new Guid(hash);
+#if NET
+        Span<byte> bytesToChecksum = stackalloc byte[16];
```

Comment:
Do we have a constant we can use for this?

Reply:
I don't think there is anything in the Guid class accessible to us. A regexp search for "guid.*16" in Roslyn has several hits, but nothing is sticking out as a good candidate to use.